public/assets/ts/admin/editor.ts

```diff
@@ -48,6 +55,23 @@ class Editor {
       .trim();
   }
 
+  private async getFileData(): Promise<any> {
+    const options = {
+      method: "GET",
+      headers: {
+        "Content-Type": "application/json",
+      },
+    };
+
+    const response = await fetch(this.sourceDataURL, options);
+
+    if (response.status == 400) {
```

Comment:
hmmm

In normal use cases, a bad request will only return a 400.  I think there might be an edge case where if someone goes to an editor page that doesn't exist (e.g. `admin/editor?page=just-trust-me-bro`) then it could return a 404.  I'll update it